std/string.d

```diff
@@ -2638,6 +2638,18 @@ auto lineSplitter(KeepTerminator keepTerm = KeepTerminator.no, Range)(Range r)
     return LineSplitter!(keepTerm, Range)(r);
 }
 
+///
+@safe pure unittest
+{
+    string s = "Hello\nmy\rname\nis";
+    string[] split = splitLines(s);
+
+    int idx = 0;
+    foreach(it; lineSplitter(s)) {
```

Comment:
`assert(splitLines(s).equal(lineSplitter(s)))` should work too, which I think illustrates how this example isn't very illustrating. The difference between the two is that `lineSplitter` is lazy and does not allocate; if this example is to use `splitLines`, I think this difference has to be shown somehow, or at least pointed out in a comment.